persistence-jpa/javadsl/src/main/java/com/lightbend/lagom/internal/javadsl/persistence/jpa/JpaReadSideImpl.java

```diff
@@ -227,5 +242,51 @@ private Query defaultBindUpdateOffsetQuery(Query query, Long sequenceOffset, Str
                     .setParameter(3, sequenceOffset)
                     .setParameter(4, timeUuidOffset);
         }
+        
+        private class BatchedOffsetUpdater {
+
+		private Long updateOffsetRequestCount = 0l;
+		private Long lastupdatedOffsetRequestCount = 0l;
+		private Offset offsetToBeUpdated = null;
+		private Object lock = new Object();
+
+		private Timer offsetUpdaterTimer = new Timer("offset-update-timer", true);
+
+		BatchedOffsetUpdater() {
+			offsetUpdaterTimer.schedule(new UpdateOffsetTask(), 5 * 1000);
+		}
+
+		public void checkAndUpdateOffset(Offset offset) {
+			synchronized (lock) {
+				updateOffsetRequestCount++;
```

Comment:
I think this should use [`groupedWithin`](https://doc.akka.io/docs/akka/current/stream/stages-overview.html?language=scala#groupedwithin) as discussed in #667 rather than a custom counter and timer task.